packages/ethereum/tasks/slimDeployments.ts

```diff
@@ -3,31 +3,38 @@ import type { DeployFunction } from 'hardhat-deploy/types'
 import { join } from 'path'
 import { readdir, writeFile } from 'fs/promises'
 
-const DEPLOYMENTS_PATH = join(__dirname, '..', 'deployments')
-
 // deployments are done by `hardhat-deploy`
 // after a deployment, `hardhat-deploy` populates `deployments`
 // folder with various artifacts, this task loops through `deployments`
 // folder and removes data that are optional & will end up being commited
-const main: DeployFunction = async function (_hre: HardhatRuntimeEnvironment) {
-  const networks = await readdir(DEPLOYMENTS_PATH)
-
-  for (const network of networks) {
-    const contracts = await readdir(join(DEPLOYMENTS_PATH, network)).then((files) =>
-      files.filter((file) => file.endsWith('.json'))
-    )
+const main: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const basePath = join(
+    __dirname,
+    '..',
+    'deployments',
+    hre.environment,
+    hre.network.name === 'hardhat' ? 'localhost' : hre.network.name
+  )
+  const contracts = (await readdir(basePath)).filter((filename: string) => filename.endsWith('.json'))
 
-    for (const contract of contracts) {
-      const filePath = join(DEPLOYMENTS_PATH, network, contract)
-      const data = require(filePath)
-      const slimmed = {
-        address: data.address,
-        transactionHash: data.transactionHash,
-        abi: data.abi
-      }
-
-      await writeFile(filePath, JSON.stringify(slimmed, null, 2))
+  for (const contract of contracts) {
+    const filePath = join(basePath, contract)
+    const data = require(filePath)
+    const contractName = contract.replace('.json', '')
+    const compilerData =
+      data.compilerData ?? (await hre.artifacts.getBuildInfo(`contracts/${contractName}.sol:${contractName}`))
+    const slimmed = {
```

Comment:
you say in the issue description we will not be slimming the data, this says slimmed? it looks slimmed to me

Reply:
It is slimmed, but before it got also slimmed when creating a fixture and it slimmed *all* contracts on *all* chains which can lead to errors